hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -94,6 +94,18 @@ class RobotHW : public InterfaceManager
 
     return in_conflict;
   }
+/** \name Mode Switching
+   *\{*/
+
+  /**
+   * Check (in non-realtime) if given controller to be started and started can be switched from the current state of the RobotHW
```

Comment:
Also, please mention that this is just a check, that the actual resource mode switch takes place in doSwitch.

You could additionally (and optionally) use Doxygen's `\sa` tag to reference `doSwitch`